src/types.ts

```diff
@@ -0,0 +1,68 @@
+// Global namespace properties needed for communicating with Rune
+export interface InitInput {
+  startGame: () => void
+  resumeGame: () => void
+  pauseGame: () => void
+  getScore: () => number
+}
+
+export interface RuneExport {
+  // External properties and functions
+  version: string
+  init: (input: InitInput) => void
+  gameOver: () => void
+  getChallengeNumber: () => number
+  deterministicRandom: () => number
+
+  // Internal properties and functions
+  _doneInit: boolean
```

Comment:
FFTI: Instead of this, should we just declare a variable, e.g. `let _doneInit: boolean = false` to align with `doneFirstPlay`?

Reply:
We'll be moving it out in another PR which will be responsible for event order validation. I tried not to touch the Rune object itself in this PR.